code/datums/components/sniper_wallpierce.dm

```diff
@@ -0,0 +1,22 @@
+/datum/component/sniper_wallpierce
+	var/pierces_left = 1 //default to 1 wall
+
+#define PROJ_PASSNONE 0
+#define PROJ_PASSWALL 1
+#define PROJ_PASSOBJ 2
```

Comment:
I'd move those to some _stdlib file (projectiles.dm?) because as they are these depend on alphabetical file order to work.